src/liballoc/rc.rs

```diff
@@ -438,6 +437,38 @@ impl Rc<str> {
     }
 }
 
+impl<T> Rc<[T]> {
```

Comment:
This may also be more easily implementable by consuming `Vec<T>` as you've got to copy data anyway. With `Vec<T>` the `box_free` also doesn't need to be exposed as you can just `.set_len(0)` to drop all the elements.